app/js/lib/foxbox/webpush.js

```diff
@@ -1,37 +1,36 @@
 'use strict';
 
+import EventDispatcher from './event-dispatcher';
+
 // Private members
 const p = Object.freeze({
   // Properties,
-  net: Symbol('net'),
+  api: Symbol('api'),
   settings: Symbol('settings'),
 
   // Methods:
   listenForMessages: Symbol('listenForMessages'),
-  dispatchEvent: Symbol('dispatchEvent'),
 });
 
-export default class WebPush {
-  constructor(network, settings, callback) {
-    this[p.net] = network;
+export default class WebPush extends EventDispatcher {
+  constructor(api, settings) {
+    super(['push-action']);
```

Comment:
Yep, we need to agree on the event namings for sure. Lower case, dashes to separate words and "preterite verb model" - sounds good?

Here we can name it as ```push-message-received``` or ```push-action-received```.